### PR TITLE
Mixed single-SignOn and form authentication

### PR DESCRIPTION
**Pull request by amergey**
Describe a way to configure Mixed SSO and form authentication with Waffle in spring-security

### CODE REVIEW

File: Docs/spring/SpringSecuritySingleSignOnFilter.md [hunk 1]
```diff
@@ -171,6 +171,56 @@ The `basicSecurityFilterProvider` bean supports a custom Basic authentication Re
     <property name="realm" value="DemoRealm" />
 </bean>
 ```
+Mixed Single-SignOn and Form
+--------------------------------------
+To support both single sign-on and form-based authentication with spring security similarly to [TomcatMixedSingleSignOnAndFormAuthenticatorValve](https://github.com/Waffle/waffle/blob/master/Docs/tomcat/TomcatMixedSingleSignOnAndFormAuthenticatorValve.md)
```
Comment (dblock) [inline Docs/spring/SpringSecuritySingleSignOnFilter.md:176]: Make this a relative link (remove github.com/....).

File: Docs/spring/SpringSecuritySingleSignOnFilter.md [hunk 2]
```diff
@@ -171,6 +171,56 @@ The `basicSecurityFilterProvider` bean supports a custom Basic authentication Re
...
+
+Split single sign-oon and form based authentication in dedicated entry point configurations
```
Comment (dblock) [inline Docs/spring/SpringSecuritySingleSignOnFilter.md:178]: Typo: OoN

File: Docs/spring/SpringSecuritySingleSignOnFilter.md (hunk 1 above)
Comment (dblock) [reply Docs/spring/SpringSecuritySingleSignOnFilter.md:176]: Sentences need a period at the end.

File: Docs/spring/SpringSecuritySingleSignOnFilter.md [hunk 3]
```diff
@@ -171,6 +171,56 @@ The `basicSecurityFilterProvider` bean supports a custom Basic authentication Re
...
+        <td><input type="submit" value="Login" /></td>
+    </tr>
+</table>
+</form>
+<hr>
+<a href="<%=request.getContextPath()%>/waffle">
+   Login (Negotiate)
+</a>
+```
+Defining the redirection after a successful single sign-on authentication can be achieved by registering a redirect servlet with "/waffle" url-mapping.
```
Comment (dblock) [inline Docs/spring/SpringSecuritySingleSignOnFilter.md:223]: Do you have an example of how to do that?

**Review by dblock — CHANGES_REQUESTED** [review]
Typos and stuff. Thanks for doing this!

File: Docs/spring/SpringSecuritySingleSignOnFilter.md (hunk 1 above)
Comment (amergey) [reply Docs/spring/SpringSecuritySingleSignOnFilter.md:176]: I used an absolute link to be consistent as all links in Docs/spring including this one are already using absolute links

File: Docs/spring/SpringSecuritySingleSignOnFilter.md (hunk 2 above)
Comment (amergey) [reply Docs/spring/SpringSecuritySingleSignOnFilter.md]: fixed

File: Docs/spring/SpringSecuritySingleSignOnFilter.md (hunk 3 above)
Comment (amergey) [reply Docs/spring/SpringSecuritySingleSignOnFilter.md]: Example added

**Review by hazendaz — APPROVED** [review]

**Comment by hazendaz** [comment]
@dblock LGTM.  The change request is still flagged, not sure if you have to do something to accept there.

File: Docs/spring/SpringSecuritySingleSignOnFilter.md (hunk 1 above)
Comment (dblock) [reply Docs/spring/SpringSecuritySingleSignOnFilter.md]: Feel free to fix the others too :)

**Pull request merged**